### PR TITLE
Add support for Hyprland

### PR DESCRIPTION
**Pull request by andresilva**
https://github.com/tecosaur/emacs-everywhere/issues/50

### CODE REVIEW

**Comment by andresilva** [comment]
Everything works except actually pasting the contents, although that's unrelated to this PR. The content is successfully copied to the clipboard. I also tried using `wtype` instead with the same result. If I do `wtype -M shift -k insert` in a terminal it works as expected (same for `ydotool`).

File: emacs-everywhere.el [hunk 1]
```diff
@@ -460,9 +461,10 @@ Please go to 'System Preferences > Security & Privacy > Privacy > Accessibility'
                (pcase emacs-everywhere--display-server
                  (`(x11 . ,_) (emacs-everywhere--app-info-linux-x11))
                  (`(wayland . KDE) (emacs-everywhere--app-info-linux-kde))
+                 (`(wayland . Hyprland) (emacs-everywhere--app-info-linux-hyprland))
                  (_ (user-error "Unable to fetch app info with display server %S" emacs-everywhere--display-server)))))
     (make-emacs-everywhere-app
-     :id (string-to-number window-id)
+     :id window-id
```
Comment (andresilva) [inline emacs-everywhere.el:467]: Hyprland window ids aren't numeric, might as well keep it as a string everywhere.

**Comment by tecosaur** [comment]
Hmm, I'm onboard with implementing specific support for Gnome, KDE, and probably wlroots directly in `emacs-everywhere.el`, however given how many more niche DEs there are, I'm not sure how far I want to go in baking in support for them.

Perhaps as a first step, I could try making the app-info function used a variable, and changing the window `:id` to always be a string.

**Comment by andresilva** [comment]
Hyprland is wlroots-based, I don't know of any way to implement this generically for  all wlroots compositors. For sway you'll need to use sway-specific tooling as well as far as I know. Hyprland is the most popular wlroots compositor.

**Comment by tecosaur** [comment]
Mmm, while it may not currently be possible to implement a wlroots-generic method, I don't think that supporting the entire set of wayland compositors is a sensible goal.

What we can do though is have say a wiki page for less common ones though, and link to it in the docstring.

**Comment by tecosaur** [comment]
I think I'll close this PR for now, but let's not waste your work! I'll start a wiki page on adding support for less-common wayland compositors, and you can paste your code there :slightly_smiling_face:

**Comment by tecosaur** [comment]
Here's a wiki page for this: https://github.com/tecosaur/emacs-everywhere/wiki/Custom-app-info-functions-for-less-common-compositors

**Comment by tecosaur** [comment]
(the wiki page should be publically editable)

**Comment by andresilva** [comment]
Thanks, I added instructions for Hyprland there.

**Comment by edrex** [comment]
There is https://wayland.app/protocols/wlr-foreign-toplevel-management-unstable-v1 supported by (most?) wlroots compositors as well as hyprland. There is a CLI at https://sr.ht/~leon_plickat/lswt/.

**Pull request closed without merging**